src/mono/browser/runtime/diagnostics/client-commands.ts

```diff
@@ -0,0 +1,442 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import type { DiagnosticCommandProviderV2 } from "../types";
+
+import { SessionId } from "./common";
+import { runtimeHelpers } from "./globals";
+
+// ADVR_V1\0
+export const advert1 = [65, 68, 86, 82, 95, 86, 49, 0,];
+// DOTNET_IPC_V1\0
+export const dotnet_IPC_V1 = [68, 79, 84, 78, 69, 84, 95, 73, 80, 67, 95, 86, 49, 0];
+
+// this file contains the IPC commands that are sent by client (like dotnet-trace) to the diagnostic server (like Mono VM in the browser)
+// just formatting bytes, no sessions management here
+
+
+export function advertise () {
+    // xxxxxxxx-xxxx-4xxx-xxxx-xxxxxxxxxxxx
+    const uuid = new Uint8Array(16);
+    crypto.getRandomValues(uuid);
+    uuid[7] = (uuid[7] & 0xf) | 0x40;// version 4
+
+    const pid = runtimeHelpers.mono_wasm_process_current_pid();
+
+    return Uint8Array.from([
+        ...advert1,
+        ...uuid,
+        ...serializeUint64([0, pid]),
+        0, 0// future
+    ]);
+}
+
+export function commandStopTracing (sessionID:SessionId) {
+    return Uint8Array.from([
+        ...serializeHeader(CommandSetId.EventPipe, EventPipeCommandId.StopTracing, computeMessageByteLength(8)),
+        ...serializeUint64(sessionID),
+    ]);
+}
+
+export function commandResumeRuntime () {
+    return Uint8Array.from([
+        ...serializeHeader(CommandSetId.Process, ProcessCommandId.ResumeRuntime, computeMessageByteLength(0)),
+    ]);
+}
+
+export function commandProcessInfo3 () {
+    return Uint8Array.from([
+        ...serializeHeader(CommandSetId.Process, ProcessCommandId.ProcessInfo3, computeMessageByteLength(0)),
+    ]);
+}
+
+
+export function commandGcHeapDump (extraProviders:DiagnosticCommandProviderV2[]) {
+    return commandCollectTracing2({
+        circularBufferMB: 256,
```

Comment:
nit: Is it worth making this configurable? I know using perfview, I've sometimes lost events because I filled this buffer. Is this buffer in wasm memory or is it in JavaScript memory?